### PR TITLE
Catch signals during initialization also

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

Right now if you Ctrl+C before initialization ends, it won't properly clean things up. For Kubernetes specially, this can be problematic as the clusters will stay there using resources of the machine.

## Proposal

Make it so that if a signal is caught during initialization, it'll halt initialization and properly clean things up before exiting the process

## Test Plan

Sanity checked regular `linera net up`: https://gist.github.com/andresilva91/58914f9dd0e76a2b41b95e615f51e07e

Tested `linera net up --kubernetes` in normal settings: https://gist.github.com/andresilva91/10e48a52e9e00f698dc6728e8eabecc2

And interrupting in the middle of the initialization: https://gist.github.com/andresilva91/1a0e78512ef238f1e6646174e75dbea4

### CODE REVIEW

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1273** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1273?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1275** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1275?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1274** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1274?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1281** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1281?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1281?utm_source=stack-comment).

File: linera-service/src/linera.rs [hunk 1]
```diff
@@ -2281,28 +2303,36 @@ async fn run(options: ClientOptions) -> Result<(), anyhow::Error> {
...
+                            testing_prng_seed: *testing_prng_seed,
+                            num_initial_validators: *validators,
+                            num_shards: *shards,
+                            binaries_dir: binaries_dir.clone(),
+                        };
+                        let (mut net, client1) = config.instantiate().await?;
+                        let result = Ok(net_up(extra_wallets, &mut net, client1).await?);
+                        listen_for_signals(&mut shutdown_receiver, &mut net).await?;
+                        result
+                    } else {
```
Comment (ma2bd) [inline linera-service/src/linera.rs:2322]: It's a bit sad to duplicate the content of the `else` branch. How about starting with something like this?
```
#[cfg(feature = "kubernetes")]
let (kubernetes, binaries_dirs) = (*kubernetes, binaries_dirs.clone());
#[cfg(not(feature = "kubernetes"))]
let (kubernetes, binaries_dirs) = (*kubernetes, PathBuf::default());
```

File: linera-service/src/linera.rs [hunk 2]
```diff
@@ -2033,9 +2033,43 @@ impl Job {
...
+
+    let _ = shutdown_sender.send(()).await;
+}
+
+async fn listen_for_signals(
+    shutdown_receiver: &mut tokio::sync::mpsc::Receiver<()>,
+    net: &mut impl LineraNet,
+) -> anyhow::Result<()> {
+    tokio::select! {
+        _ = shutdown_receiver.recv() => {
```
[the remark this answers is not in the document]
Comment (andresilva91) [reply linera-service/src/linera.rs:2061]: 🤣 my bad, forgot to remove it!

File: linera-service/src/linera.rs (hunk 1 above)
Comment (andresilva91) [reply linera-service/src/linera.rs:2322]: Ah, true, good idea! Will do

**Pull request merged**